shared/clientstats/scrapers.go

```diff
@@ -0,0 +1,197 @@
+package clientstats
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/prom2json"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+func scrapeProm(url string) (map[string]*dto.MetricFamily, error) {
+	mfChan := make(chan *dto.MetricFamily, 1024)
```

Comment:
Any reason for the 1024 instead of just `1`? Do we envision this channel needing to have a large buf?

Reply:
Good catch, this shouldn't be buffered at all and was hiding synchronous blocking behavior. Replaced this with a goroutine -- a little bit complex to follow due to error handling but still simpler than the equivalent code would be without the prom2json library method.